tox.ini

```diff
@@ -0,0 +1,112 @@
+[tox]
+envlist =
+    py37-test{,-astropy30, -astropy31, -astropy32}
+    py38-test{,-astropy32}
+    py39-test
+requires =
+    setuptools >= 30.3.0
+    pip >= 19.3.1
+    tox-pypi-filter >= 0.12
+
+[testenv]
+pypi_filter = https://raw.githubusercontent.com/astropy/ci-helpers/master/pip_pinnings.txt
+commands =
+    pip freeze
+    python setup.py test
+
+description =
+    run tests
+    astropy30: with astropy 3.0.*
+    astropy31: with numpy 3.1.*
+    astropy32: with numpy 3.2.*
+
+deps =
+    #-rpip-requirements
+    requests
+    mimeparse
+    pytest-astropy
+    requests_mock
+
+    astropy30: astropy==3.0.*
+    astropy31: astropy==3.1.*
+    astropy32: astropy==3.2.*
+    astropydev: git+https://github.com/astropy/astropy.git#egg=astropy
+
+#    # Linkcheck is broken with cryptic error with Sphinx 3.1.0
+#    # https://github.com/astropy/astropy/issues/10489
+#    # It also emits a warning from BLS docstring that we cannot banish with numpydoc 1.1.0
+    linkcheck: sphinx<3.1
+    linkcheck: numpydoc<1.1.0
+
+[testenv:egg_info]
+description = ensure egg_info works without dependencies
+commands =
+    pip freeze
+    python setup.py egg_info
+
+[testenv:cov]
+description = determine the code coverage
+deps:
+     # had to pin this
+    coverage==4.5.4
+    #-rpip-requirements
+    requests
+    mimeparse
+    pytest-astropy
+    requests_mock
+
+commands =
+    pip freeze
+    python setup.py test --coverage
```

Comment:
at some point running tests with setup.py can be retired, too, in favour of directly running pytest, but it's OK for now.